asf_heat_pump_suitability/pipeline/suitability/calculate_suitability.py

```diff
@@ -258,3 +263,35 @@ def compute_dict_lsoa_suitability_scores(df: pl.DataFrame, lsoa: str) -> dict:
     scores_dict["n_properties"] = len(df)
 
     return scores_dict
+
+
+def aggregate_dict_lsoa_suitability_and_features(
+    df: pl.DataFrame, lsoa_code: str
+) -> dict:
+    """
+    Compute single aggregate weighted score for one LSOA and weight and aggregate features to provide LSOA-level
+    averages.
+
+    Args:
+        df (pl.DataFrame): EPC dataset with [un]weighted suitability scores per EPC record
+        lsoa_code (str): LSOA code to aggregate for
+
+    Returns:
+        dict: [un]weighted average suitability scores for one LSOA for each tech type with average features for the LSOA
+    """
+    # Filter df to specified LSOA
+    lsoa_df = df.filter(pl.col("lsoa") == lsoa_code)
+
+    # Compute weighted score for each property in one LSOA - this dataset will be one row per property in a single LSOA
+    lsoa_df = compute_df_weighted_score(lsoa_df)
+
+    # Weight and aggregate the features to LSOA-level for one LSOA
+    lsoa_df = aggregate_features.extend_df_feature_weight(lsoa_df)
+    lsoa_df = epc.convert_df_epc_rating(lsoa_df)
+    lsoa_features_dict = aggregate_features.aggregate_dict_features_per_lsoa(lsoa_df)
```

Comment:
took me a while to get my head around the 'feature_weight' and 'total_weight' idea, but think this is a really elegant way to deal with it!!